package/base-files/files/etc/uci-defaults/12_network-generate-ula

```diff
@@ -1,11 +1,7 @@
 [ "$(uci -q get network.globals.ula_prefix)" != "auto" ] && exit 0
 
-r1=$(dd if=/dev/urandom bs=1 count=1 |hexdump -e '1/1 "%02x"')
-r2=$(dd if=/dev/urandom bs=2 count=1 |hexdump -e '2/1 "%02x"')
-r3=$(dd if=/dev/urandom bs=2 count=1 |hexdump -e '2/1 "%02x"')
-
 uci -q batch <<-EOF >/dev/null
-	set network.globals.ula_prefix=fd$r1:$r2:$r3::/48
+	set network.globals.ula_prefix="$(hexdump -vn 5 -e '"fd" 1/1 "%02x:" 2/2 "%02x:"' /dev/urandom):/48"
```

Comment:
The `2/2 "%02x:"` part should be either `2/2 "%x:"` instead, for compressed IPv6 notation (preferred, e.g. `fd01:2:345::/48`) or `2/2 "%04x:"` for uncompressed notation (`fd01:0002:0345::/48`).